### DISCUSSION TITLE
Help: get Type interface for ctx

### DISCUSSION THREAD

**Discussion by aleph65** [telegraf/telegraf#1843] [Q&A]
How can I get the Type for this ctx object?

bot.on(message('text'), async (ctx) => {

I understand it is implicitly typed and Typescript picks up on suggestions when inside the actual .on function...

But I'd like to pass this ctx object to a different function. If I type it as ctx: Context, it doesn't work (missing some things). If I pass to other funtion I have to put type of ctx as "any" which I'd like to avoid

Help plz?

**Comment by MKRhere** [comment]
Sorry for the delay. The correct type is as follows:

```TS
import type { Context } from "telegraf";
import type { Update, Message } from "telegraf/types";

type TextMessageContext = Context<Update.MessageUpdate<Message.TextMessage>>;
```

You can also extract the guarded type from a filter:

```TS
type Guarded<T> = T extends (x: any) => x is infer T ? T : never;

const filter = message("text");
type GuardedCtx = Guarded<typeof filter>;
```
